internal/api/v1/admin/paymails/paymails_api_test.go

```diff
@@ -0,0 +1,177 @@
+package paymails_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/commands"
+	"github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/admin/paymails/paymailstest"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/spvwallettest"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPaymailsAPI_DeletePaymail(t *testing.T) {
+	id := "xpub22e6cba6-ef6e-432a-8612-63ac4b290ce9"
+	tests := map[string]struct {
+		responder        httpmock.Responder
+		expectedResponse *response.PaymailAddress
+		expectedErr      error
+	}{
+		fmt.Sprintf("HTTP DELETE /api/v1/admin/paymails/%s response: 200", id): {
+			expectedResponse: paymailstest.ExpectedCreatedPaymail(t),
+			responder:        httpmock.NewStringResponder(http.StatusOK, http.StatusText(http.StatusOK)),
+		},
+		fmt.Sprintf("HTTP DELETE /api/v1/admin/paymails/%s response: 400", id): {
+			expectedErr: spvwallettest.NewBadRequestSPVError(),
+			responder:   httpmock.NewJsonResponderOrPanic(http.StatusBadRequest, spvwallettest.NewBadRequestSPVError()),
+		},
+		fmt.Sprintf("HTTP DELETE /api/v1/admin/paymails/%s response: 500", id): {
+			expectedErr: spvwallettest.NewInternalServerSPVError(),
+			responder:   httpmock.NewJsonResponderOrPanic(http.StatusBadRequest, spvwallettest.NewInternalServerSPVError()),
```

Comment:
The reason why we have two test cases for HTTP status code: 500 is the SPV Wallet API can return for such code two type of responses. It can return the JSON response (SPVErrror structure) or string response depending on different circumstances.